File: origin/YuGiOhCard/Controller/ControllerUtils/PushUtils.m
#import "PushUtils.h"
#import "CardViewController.h"

@implementation PushUtils

+(void) pushCard: (CardItem *) item navController: (UINavigationController *) nav {
    UIStoryboard * cardStory = [UIStoryboard storyboardWithName:@"CardStory" bundle:nil];
    CardViewController * cc = [cardStory instantiateViewControllerWithIdentifier:@"cardViewController"];
    [cc setCardId:item.card_id];
    [cc setCardName:item.name];
    [nav pushViewController:cc animated:YES];
}

@end

File: swift/YuGiOhCard/Controller/ControllerUtils/PushUtils.swift
import UIKit

class PushUtils: NSObject {

    class func pushCard(item: CardItem, navController: UINavigationController) {
        let cardStory = UIStoryboard(name: "CardStory", bundle: nil)
        let cc = cardStory.instantiateViewControllerWithIdentifier("cardViewController") as! CardViewController
        cc.cardId = item._id
        cc.cardName = item.name
        navController.pushViewController(cc, animated: true)
        
    }

}
